packages/aws-rfdk/lib/deadline/lib/version.ts

```diff
@@ -72,6 +72,106 @@ abstract class VersionQueryBase extends Construct implements IVersion {
   }
 }
 
+/**
+  * This class is reposonsible to do basic operations on version format.
+  */
+export class Version {
```

Comment:
It looks like this class could implement the `IPatchVersion` interface here. We could leverage [TypeScript accessors](https://www.typescriptlang.org/docs/handbook/classes.html#accessors) for implementing the individual components, e.g.:

```ts
export class Version {
  // ...
  public get majorVersion(): number {
    return this.components[0];
  }
}
```

This could simplify the code in `VersionQuery.parseVersionString`.

Reply:
updated.